content/blog/adopting-existing-cloud-resources-into-pulumi/index.md

```diff
@@ -0,0 +1,242 @@
+
+We'll review referencing existing resources, and then dive deeper into how you can adopt existing resources with Pulumi.
+
+## Referencing Existing Resources
+For referencing existing resoruces, Pulumi offers several tools.  
+
+* The `.get` methods available on every resource let you [get all the details for a resource](https://www.pulumi.com/docs/reference/pkg/nodejs/pulumi/aws/ec2/#Vpc-get) from the cloud provider based just on it's `id`.
+* The `StackReference` resource lets you reference outputs of another stack for use as inputs to another stack, which is very useful for [organizing projects and stack](https://www.pulumi.com/docs/reference/organizing-stacks-projects/). 
+* The [`terraform.state.RemoteStateReference()`](https://www.pulumi.com/blog/using-terraform-remote-state-with-pulumi/), [`aws.cloudformation.getStack()`](https://www.pulumi.com/docs/reference/pkg/nodejs/pulumi/aws/cloudformation/#getStack) and [`azure.core.TemplateDeployment.get()`](https://www.pulumi.com/docs/reference/pkg/nodejs/pulumi/azure/core/#TemplateDeployment-get) let you reference outputs from existing Terraform, CloudFormation and ARM deployments respectively.
+
+Together, these make it easy to reference existing infrastructure regardless of how it was provisioned.
+
+## Adopting Existing Resources
+
+For adopting resources, Pulumi offers the [`import`](https://www.pulumi.com/docs/reference/pkg/nodejs/pulumi/azure/core/#TemplateDeployment-get) resource option to request that a resource defined in your Pulumi program adopt an existing resource in the cloud provider instead of creating a new one.  In keeping with it's focus on infrastructure as *code*, Pulumi lets you specify this `import` behaviour inside the Pulumi code for you infrastructure deployment, instead of outside it in some manual workflow.  In it's simplest form looks like this:
+
+```ts
+const myVpc = new aws.ec2.Vpc("my-vpc", {
+    cidrBlock: "10.11.0.0/16",
+}, { import: "vpc-02cd0cd2d807f340c"});
+```
+
+When Pulumi first sees a resource (in this case `my-vpc`), if it has an `import` option set, it will adopt the existing resource specified by that `import` ID from the cloud provider under management instead of creating a new resource.
+
+We can see this when we deploy our Pulumi program with `pulumi up`:
+
+```
+$ pulumi up
+Previewing update (dev):
+
+     Type                 Name               Plan       
+     pulumi:pulumi:Stack  importtestaws-dev             
+ =   └─ aws:ec2:Vpc       my-vpc             import     
+ 
+Resources:
+    = 1 to import
+    1 unchanged
+```
+
+Because the `import` is provided in code, it can be configured in many different ways.  We've seen users look up import ids from Pulumi config instead of hardcoding into their programs, construct them from predicatable names based on existing program parameters, or even conditionally add the `import: <id>` property depending on whether they want to deploy a stack in "adopt" mode or in "create" mode.
```

Comment:
```suggestion
Because the `import` is provided in code, it can be configured in several different ways.  You can look up import ids from Pulumi config instead of hardcoding them into programs, construct ids from predictable names based on program parameters, or even conditionally add the `import: <id>` property depending on whether one wants to deploy a stack in "adoption" or "creation" mode.
```